src/widgets/user-profile-modal/model/useUserProfileData.ts

```diff
@@ -0,0 +1,23 @@
+import { useUserQuery } from '@features/user/get-single-user'
+import { usePostsUI } from '@shared/store/posts-ui'
+
+export const useUserProfileData = () => {
+  const { selectedUserId, showUserModal, setShowUserModal, setSelectedUserId } = usePostsUI()
+
+  // ID로 사용자 데이터 조회
+  const { data: selectedUser, isLoading } = useUserQuery(selectedUserId!, {
+    enabled: !!selectedUserId && showUserModal,
+  })
+
+  const handleCloseModal = () => {
+    setShowUserModal(false)
+    setSelectedUserId(null)
+  }
+
+  return {
+    selectedUser,
+    isLoading,
+    showUserModal,
+    handleCloseModal,
+  }
```

Comment:
as const 를 붙여야 타입이 자동으로 잘 완성되더라구여
```suggestion
  } as const
```